service/worker/scheduler/workflow.go

```diff
@@ -162,7 +172,7 @@ var (
 	currentTweakablePolicies = tweakablePolicies{
 		DefaultCatchupWindow:              365 * 24 * time.Hour,
 		MinCatchupWindow:                  10 * time.Second,
-		RetentionTime:                     7 * 24 * time.Hour,
+		RetentionTime:                     0, // TODO: enable later: 7 * 24 * time.Hour,
```

Comment:
Because we want to release 1.21.5 with the new logic present but disabled for that change as well. The alternative is to leave this alone here (on master) and make this change on the release branch after cherry-picking this. But we prefer not to make changes on the release branch directly.

I guess it could also be in a separate PR, but that's just more noise